File: Skornel02.ETR.Common/Dtos/ExamUpdateDto.cs
using Skornel02.ETR.Common.Enums;

namespace Skornel02.ETR.Common.Dtos;

public record ExamUpdateDto(
    string CourseCode,
    string CourseSemester,
    DateTimeOffset OriginalStart,
    DateTimeOffset End,
    ExamType ExamType,
    string ClassRoomAddress,
    string ClassRoomNumber
);

File: Skornel02.ETR.Web/src/schemas/ExamUpdateDto.ts
import { z } from "zod";
import { ExamType } from "../enums/examtypes";

export const ExamUpdateDtoSchema = z.object({
    courseCode: z.string(),
    courseSemester: z.string(),
    originalStart: z.string(),
    end: z.string(),
    examType: z.nativeEnum(ExamType),
    classRoomAddress: z.string(),
    classRoomNumber: z.string()

}).superRefine(({ originalStart, end }, ctx) => {
    if (new Date(originalStart).getTime() >= new Date(end).getTime()) {
        ctx.addIssue({
            code: 'custom',
            path: ['end'],
            message: 'A vége nem lehet a kezdete előtt!'
        });
    }
});

export type ExamUpdateDto = z.infer<typeof ExamUpdateDtoSchema>;
